apollos-church-api/src/data/interactions/index.js

```diff
@@ -0,0 +1,54 @@
+import { Interactions } from '@apollosproject/data-connector-rock';
+import gql from 'graphql-tag';
+
+export class dataSource extends Interactions.dataSource {
+  async createPageViewInteraction({ pageTitle, pageUrl }) {
+    const {
+      dataSources: { RockConstants, Auth },
+    } = this.context;
+
+    const interactionComponent = await RockConstants.createOrFindInteractionComponent(
+      {
+        componentName: pageTitle,
+        channelId: 3, // ExternalWebsite
+        entityId: null,
+      }
+    );
+
+    const currentUser = await Auth.getCurrentPerson();
+
+    await this.post('/Interactions', {
+      PersonAliasId: currentUser.primaryAliasId,
+      InteractionComponentId: interactionComponent.id,
+      Operation: 'View',
+      InteractionDateTime: new Date().toJSON(),
+      InteractionSummary: pageTitle,
+      InteractionData: pageUrl,
+      // We can introduce the session variable if the client asks for it.
+      // I ran into some problems with stale sessions causing this request to fail
+      // So I figured it would be better to default this to off.
+      // InteractionSessionId: this.context.sessionId,
```

Comment:
not a big deal, our `sessionId` is misleading I think because we only gen on auth token regen